### PR TITLE
Package for public repo

### PR DESCRIPTION
**Pull request by kevinhcolgan**
Add some features to make packaging for a public repo better:
* tsdx for building
* size-limit check
* tsconfig update
* updated README

### CODE REVIEW

**Review by flipscholtz — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
+    "prepare": "tsdx build",
+    "script": "ts-node -O '{\"module\":\"commonjs\"}' --transpile-only --files",
+    "size": "size-limit",
+    "analyze": "size-limit --why"
   },
+  "module": "dist/prove-ethereum-wallet.esm.js",
+  "size-limit": [
+    {
+      "path": "dist/prove-ethereum-wallet.cjs.production.min.js",
+      "limit": "200 KB"
```
Comment (rado0x54) [inline package.json:30]: Is that a sensible limit? I feel it should be much smaller for being such a thing lib.

File: package.json [hunk 2]
```diff
@@ -1,39 +1,58 @@
...
   "dependencies": {
     "@ethersproject/abstract-signer": "^5.5.0",
-    "bignumber.js": "^9.0.1",
-    "es6-promisify": "^7.0.0",
     "ethers": "^5.5.1",
-    "rambda": "^6.9.0",
-    "tslib": "^2.3.1",
-    "web3": "^1.6.1",
-    "yargs": "^17.2.1"
+    "web3": "^1.6.1"
```
Comment (rado0x54) [inline package.json:40]: Can you tell me where this is still used? What complains if you remove it.

**Review by rado0x54 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (rado0x54) [reply package.json:30]: thin lib.

Comment (kevinhcolgan) [reply package.json:30]: you're right, I regenerated once I removed all the dependencies and it's much smaller, so I adjusted the limits:
```
$ size-limit
✔ Adding to empty esbuild project
✔ Adding to empty webpack project
  
  dist/prove-ethereum-wallet.cjs.production.min.js
  Size limit: 110 kB
  Size:       100.38 kB with all dependencies, minified and gzipped
  
  dist/prove-ethereum-wallet.esm.js
  Size limit: 10 kB
  Size:       2.42 kB with all dependencies, minified and gzipped
  
✨  Done in 6.10s.
```

File: package.json (hunk 2 above)
Comment (kevinhcolgan) [reply package.json]: I removed it, thanks

File: package.json (hunk 1 above)
Comment (rado0x54) [reply package.json]: that's the esm size i expected! :)

**Pull request merged**